### PR TITLE
version: export tweak version in version.h and version string

### PR DESCRIPTION
**Pull request by tejlmand**
The PR exports TWEAK version to code through version.h. This aligns the symbols available in code with those available in CMake and Kconfig.

It also introduces a new extended version string containing all version info fields, including tweak.

The new define and string is available for both KERNEL, APP, and custom version types (such as MCUBOOT).

### CODE REVIEW

**Review by pdgendt — DISMISSED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]
Doc update needed in `doc/build/version/index.rst`

**Comment by tejlmand** [comment]
> Doc update needed in `doc/build/version/index.rst`

Fixed.

**Comment by pdgendt** [comment]
Does the tweak version need to be added to "version binary descriptors" in `subsys/bindesc/Kconfig.version`? Can be done in a separate PR probably.

**Review by nordicjm — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

**Comment by tejlmand** [comment]
> Does the tweak version need to be added to "version binary descriptors" in `subsys/bindesc/Kconfig.version`? Can be done in a separate PR probably.

Probably a good idea, especially as tweak has always been present in `VERSION` file, but as this PR only touched handling of VERSION file and making values available in various contexts, then I believe extending bindesc warrants its own PR, especially as there seems to be a bit more to it, than just adding a Kconfig setting.

File: doc/build/version/index.rst [hunk 1]
```diff
@@ -73,85 +73,105 @@ To use the version information in application code, the version file must be inc
...
+|                             |                 | ``VERSION_MINOR``, |br|                           |                  |
+|                             |                 | ``PATCHLEVEL``, |br|                              |                  |
+|                             |                 | ``EXTRAVERSION``                                  |                  |
++-----------------------------+-----------------+---------------------------------------------------+------------------+
+| APP_VERSION_EXTENDED_STRING | String          | ``VERSION_MAJOR``, |br|                           | 1.2.3-unstable+4 |
+|                             |                 | ``VERSION_MINOR``, |br|                           |                  |
+|                             |                 | ``PATCHLEVEL``, |br|                              |                  |
+|                             |                 | ``EXTRAVERSION``, |br|                            |                  |
+|                             |                 | ``PATCHLEVEL``                                    |                  |
```
Comment (kkasperczyk-no) [inline doc/build/version/index.rst:173]: Shouldn't it be
```suggestion
|                             |                 | ``VERSION_TWEAK``                                    |                  |
```
?

Comment (tejlmand) [reply doc/build/version/index.rst:173]: it sure should.

Comment (tejlmand) [reply doc/build/version/index.rst]: fixed

**Review by LuDuda — APPROVED** [review]
Looks awesome 👍 

A bit unrelated, but this line does not look correct: https://github.com/zephyrproject-rtos/zephyr/pull/68160/files#diff-4919a7623a2d331ff10447d41f4d1f705af5f6962120735553687400f5bd08fbL163

**Review by nordicjm — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**